### DISCUSSION TITLE
min/maxLength not transfering to existing form

### DISCUSSION THREAD

**Discussion by lifthus** [react-hook-form/react-hook-form#9829] [Q&A] [answered]
I built an input tag component nested in styled-components div as described on "Integrating an existing form " at "Get started" of official documentation.

watch(), onChange handler etc everything is working but min/maxLength.

I didn't try but I think it will work when it comes to validation though.
but it doesn't restrict the length of input field at all.

I am using React.forwardRef and I give {...register("id")} to my own div component as described on doc.
I transfer onChange handler to nested input tag's props and it works.
but when I transfer min/maxLength, I can write infinite letters no matter how the maxLength is set.

actually I checked the transferred min/maxLength but they are undefined. 
definitely it seems they can't be used like onChange prop.

is it impossible to restrict the nested input field to have maxLength in this method?
Do I have to transfer register() itself or reveal the input tag to the place where the form is?

🙏😭

**Comment by Moshyfawn** [comment, marked as answer]
There's a `shouldUseNativeValidation` flag on `useForm`, which you can set to `true` for the `register` method to return the constraint validation properties

**Reply by lifthus** [reply]
Thank you for quick response! if you don’t mind can I ask why that flag is required in this case, even though it is automatically applied when using register() on the input tag directly?

**Reply by Moshyfawn** [reply]
`shouldUseNativeValidation` enables `register` to return extra constrain validation attributes like `max`, `minLength`, `required`. It's `false` by default, so the only props returned from the `register` method are the handlers like `onChange`, `onBlur` etc.

The attributes aren't auto-applied without the flag as they're related to native browser validation only. Unless, you're seeing a different behaviour :D

**Reply by lifthus** [reply]
Ah you are right. I think I had mistaken after giving maxLength prop to input tag directly. By the way, It was so helpful. I appreciate! 🤩

**Reply by Moshyfawn** [reply]
No problem!
